libbeat/cfgfile/reload.go

```diff
@@ -69,23 +69,19 @@ type Reload struct {
 	Enabled bool          `config:"enabled"`
 }
 
-// RunnerFactory is used for creating of new Runners
+// RunnerFactory is used for validating generated configurations and creating
+// of new Runners
 type RunnerFactory interface {
-	Create(p beat.Pipeline, config *common.Config, meta *common.MapStrPointer) (Runner, error)
+	ConfigChecker
+	Create(p beat.PipelineConnector, config *common.Config, meta *common.MapStrPointer) (Runner, error)
 }
 
-// ConfigChecker is usually combined with a RunnerFactory for implementations that can check a config
-// without a pipeline and metadata.
+// ConfigChecker is usually combined with a RunnerFactory for implementations
```

Comment:
Are there any uses of `ConfigChecker` without `RunnerFactory`? The only references I can find are casts of `RunnerFactory` to `ConfigChecker`, which should be redundant with this change. Can we move `CheckConfig` into `RunnerFactory` and remove `ConfigChecker` entirely, or is it still needed for something?

Reply:
will have a look and remove casts.